CRM/Financial/BAO/Payment.php

```diff
@@ -346,4 +346,125 @@ public static function recordRefundPayment($contributionId, $trxnData, $updateSt
     return $financialTrxn;
   }
 
+  /**
+   * @param int $contributionId
+   * @param array $trxnData
+   * @param int $participantId
+   *
+   * @return \CRM_Core_BAO_FinancialTrxn
+   */
+  public static function recordPayment($contributionId, $trxnData, $participantId) {
+    $getInfoOf['id'] = $contributionId;
+    $defaults = array();
+    $contributionDAO = CRM_Contribute_BAO_Contribution::retrieve($getInfoOf, $defaults);
+    if (!$participantId) {
+      $participantId = CRM_Core_DAO::getFieldValue('CRM_Event_DAO_ParticipantPayment', $contributionId, 'participant_id', 'contribution_id');
+    }
+
+    // load related memberships on basis of $contributionDAO object
+    // @todo - this is done in the function that completes payments so it's being done twice.
+    $contributionDAO->loadRelatedMembershipObjects();
+
+    // build params for recording financial trxn entry
+    $params['contribution'] = $contributionDAO;
+    $params = array_merge($defaults, $params);
+    $params['skipLineItem'] = TRUE;
+    $trxnsData['trxn_date'] = !empty($trxnsData['trxn_date']) ? $trxnsData['trxn_date'] : date('YmdHis');
+    $params['payment_instrument_id'] = CRM_Utils_Array::value('payment_instrument_id', $trxnsData, CRM_Utils_Array::value('payment_instrument_id', $params));
+
+    $paidStatus = CRM_Core_PseudoConstant::getKey('CRM_Financial_DAO_FinancialItem', 'status_id', 'Paid');
+    $arAccountId = CRM_Contribute_PseudoConstant::getRelationalFinancialAccount($contributionDAO->financial_type_id, 'Accounts Receivable Account is');
+    $completedStatusId = CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'contribution_status_id', 'Completed');
+
```

Comment:
done

Reply:
@eileenmcnaughton please remove this empty new line as caught in ```style warnings```